retry/retry.go

```diff
@@ -84,22 +89,26 @@ func (r Retry) waitTime(attempt uint) int {
 	return WaitTime(attempt, r.random)
 }
 
-// WaitTime calculates the wait time based off the attempt number based off
-// exponential backoff with a random delay.
+// WaitTime calculates the wait time in nanoseconds based off the attempt number based off
+// exponential backoff with a random delay. It caps at the constant maxWaitTime.
 func WaitTime(attempt uint, random *rand.Rand) int {
```

Comment:
Add unit to the name please, return type is an int so it is not clear what the unit is when we use it in the code later.
Even better, can we use a type more suitable for time instead of returning an int?

Reply:
I'll change the name, and look into refactoring the WaitTime to use a duration